sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -250,9 +248,7 @@ private boolean validateFields(List<String> fields) {
                 if (accumulator == null) {
                     accumulator = new HashSet<String>(childIds);
                 } else {
-                    if (childIds != null) {
-                        accumulator = Sets.union(accumulator, childIds);
-                    }
+                    accumulator = Sets.union(accumulator, childIds);
```

Comment:
Separated to its own commit https://github.com/cloudant/sync-android/commit/cfcced806804a1b16573bce95252d7595ba4039d